cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -571,6 +575,21 @@ public void initFeatures() {
             this.pendingInitFeaturesCall = true;
             return;
         }
+        IntentFilter intentFilter = new IntentFilter(Constants.DEVICE_ID_INTENT);
```

Comment:
put this intent/broadcaster stuff into a function

Reply:
all of this into a function
![image](https://user-images.githubusercontent.com/99720107/194527211-549368bb-94fe-4356-8f07-87f8bad59896.png)

Reply:
```
    private void registerReceiver() {
        IntentFilter intentFilter = new IntentFilter(Constants.DEVICE_ID_ACTION_KEY);
        if (intentFilter != null) {
            context.registerReceiver(deviceIdBroadcastReceiver, intentFilter);
        }

        SharedPreferences sharedPreferences = getSharedPreferences(getContext());
        if (sharedPreferences.getString(CleverPushPreferences.DEVICE_ID, null) == null) {
            sharedPreferences.edit().putString(CleverPushPreferences.DEVICE_ID, UUID.randomUUID().toString()).apply();
        }
    }

```